# HG changeset patch
# User Lachlan Andrew <lachlanbis@gmail.com>
# Date 1458028928 -39600
#      Tue Mar 15 19:02:08 2016 +1100
# Branch stable
# Node ID 04e0598f8b94f89e3bf1e4f6dd5c3d3503e350f6
# Parent  882cb9f032cef55267f4c20a94f383c5eebd8200
Allow strsplit to use '+' as a delimiter (bug #47403).

* strsplit.m: Add '+' to regexprep call to escape characters that have special
meaning in regexp.

diff --git a/scripts/strings/strsplit.m b/scripts/strings/strsplit.m
--- a/scripts/strings/strsplit.m
+++ b/scripts/strings/strsplit.m
@@ -191,17 +191,17 @@ function [cstr, matches] = strsplit (str
 
   if (is_simple)
     if (iscellstr (del))
       del = cellfun (@do_string_escapes, del, "uniformoutput", false);
     else
       del = do_string_escapes (del);
     endif
     ## Escape characters which have a special meaning in regexp.
-    del = regexprep (del, '([{}()[\]^$.*?|\\])', '\\$1');
+    del = regexprep (del, '([{}()[\]^$.*?+|\\])', '\\$1');
   endif
 
   if (isempty (str))
     cstr = {str};
   else
     if (iscellstr (del))
       del = sprintf ("%s|", del{:});
       del(end) = [];
@@ -296,16 +296,19 @@ endfunction
 %! assert (strsplit ("a", "a"), {"", ""})
 %! assert (strsplit ("aa", "a"), {"", ""})
 %! assert (strsplit ("aaa", "a"), {"", ""})
 
 ## Bug #44641
 %!assert (strsplit ("xxx<yyy", "<"), {"xxx", "yyy"})
 %!assert (strsplit ('xxx\yyy', '\'), {"xxx", "yyy"})
 
+## Bug #47403
+%!assert (strsplit ('xxx+yyy', '+'), {"xxx", "yyy"})
+
 ## Test input validation
 %!error strsplit ()
 %!error strsplit ("abc", "b", true, 4)
 %!error <invalid parameter name, 'foo'> strsplit ("abc", "b", "foo", "true")
 %!error <S and DEL must be string values> strsplit (123, "b")
 %!error <COLLAPSEDELIMITERS must be a scalar value> strsplit ("abc", "def", "collapsedelimiters", ones (3,3))
 %!error <Invalid DELIMITERTYPE> strsplit ("abc", "b", "delimitertype", "foobar")
 
